driver-core/src/main/java/com/datastax/driver/core/ExecutionInfo.java

```diff
@@ -87,7 +88,10 @@ ExecutionInfo withIncomingPayload(Map<String, ByteBuffer> incomingPayload) {
      * {@link com.datastax.driver.core.policies.RetryPolicy} may retry the
      * query on the same host, so the same host might appear twice.</li>
      * <li>if {@link com.datastax.driver.core.policies.SpeculativeExecutionPolicy speculative executions}
-     * are enabled, other hosts might have been tried speculatively as well.</li>
+     * are enabled, this will also contain hosts that were tried by other executions (however, note that
+     * this only contains hosts for which a response was received; if an execution is waiting for a response
```

Comment:
This is misleading: there is only ever at most one response received (either a result or an error); other tried hosts, if any, can only be listed here if they timed out with no response at all. I suggest: "this only contains the queried host – i.e. the host who replied with the final response – and hosts that failed to reply before the request timed out; if an execution is waiting..."